Add better logging for apollo client errors

nikomakela opened the pull request:
HH-264.
- Log errors of the graphql-client by using graphql-client-logger. Use graphql-client-logger for errors given by the error link.
- Fix the issues when an empty or wrongly created CMS-page has a bad list of modules and
the page render fails while collecting the page's for the render process.
- remove unused imports

New logging when a connection to a server is lost:
```
event - compiled client and server successfully in 344 ms (2589 modules)
11:32:25 [error] [graphql-client] - FetchError: request to http://localhost:4000/ failed, reason: connect ECONNREFUSED 127.0.0.1:4000
11:32:25 [error] [staticGeneration] - Error while generating a page:
```

Fixed the getCollections issue:
```
error - TypeError: Cannot use 'in' operator to search for 'title' in null
    at /Users/nikojmakela/Code/ratkaisutoimisto/events-helsinki-monorepo/node_modules/react-helsinki-headless-cms/cjs/archiveSearchPageContent.module-02c491b5.js:2110:22
```

nikomakela committed:
feat(logging): write error message of issues happening in apollo client

HH-264.
Log errors of the graphql-client by using graphql-client-logger.
Use graphql-client-logger for errors given by the error link.

nikomakela committed:
chore: remove unnecessary import statements

nikomakela committed:
fix: the translation types of the headless-cms service

terovirtanen commented on apps/events-helsinki/src/common-events/utils/headless-cms/headlessCmsUtils.tsx:
```diff
@@ -140,7 +140,8 @@ export const getDefaultCollections = (
 ) =>
-  getCollections(page?.modules ?? [], true)?.reduce(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getCollections(page?.modules?.filter((m) => !!m) ?? [], true)?.reduce(
```
some compilation errors caused by this line

terovirtanen reviewed (changes requested):
sonarcloud report errors:
```
src/common-events/utils/headless-cms/headlessCmsUtils.tsx(144,41): error TS7006: Parameter 'm' implicitly has an 'any' type.
src/common-events/utils/headless-cms/service.tsx(43,35): error TS7006: Parameter 'translation' implicitly has an 'any' type.
src/common-events/utils/headless-cms/service.tsx(91,35): error TS7006: Parameter 'translation' implicitly has an 'any' type.

```

nikomakela committed:
fix(cms): getCollection should not be called with null-modules

Fix the issues when an empty or wrongly created CMS-page has a bad list of modules and
the page render fails while collecting the page's for the render process.

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-96-hh-264-apollo-error-logs.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-96-hh-264-apollo-error-logs.dev.hel.ninja** :rocket::rocket::rocket:
